File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    answers = []
    answer = ''
    num = {'zero': 0, 'one': 1, 'two': 2, 'three': 3, 'four': 4, 'five': 5, 'six': 6, 'seven': 7, 'eight': 8, 'nine': 9}
    
    tmp = ''
    
    for i in range(len(s)):
        if tmp in num.keys():
            answers.append(num[tmp])
            tmp = ''
        
        if not s[i].isalpha():
            answers.append(int(s[i]))
        else:
            tmp += s[i]
    
    if tmp in num.keys():
        answers.append(num[tmp])
        
    return int(''.join(map(str, answers)))

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.*;

class Solution {
    public int solution(String s) {
        String answer = "";
        String tmp = "";
        Map<String, Integer> map = new HashMap<>();
        
        map.put("zero", 0);
        map.put("one", 1);
        map.put("two", 2);
        map.put("three", 3);
        map.put("four", 4);
        map.put("five", 5);
        map.put("six", 6);
        map.put("seven", 7);
        map.put("eight", 8);
        map.put("nine", 9);
        
        for (int i=0; i<s.length(); i++) {
            char ch = s.charAt(i);
            if (Character.isDigit(ch)) {
                answer+=ch;
                tmp = "";
                continue;
            }
            
            tmp += ch;
            if (map.containsKey(tmp)) {
                answer += map.get(tmp);
                tmp = "";
            }
        }
        if (answer == "") return 0;
        return Integer.parseInt(answer);
    }
}